api/http/handler/edgestacks/edgestack_update.go

```diff
@@ -142,50 +121,20 @@ func (handler *Handler) updateEdgeStack(tx dataservices.DataStoreTx, stackID por
 		return nil, httperror.BadRequest("edge stack with config do not match the environment type", nil)
 	}
 
-	if payload.DeploymentType == portainer.EdgeStackDeploymentCompose {
-		if entryPoint == "" {
-			entryPoint = filesystem.ComposeFileDefaultName
-		}
-
-		_, err := handler.FileService.StoreEdgeStackFileFromBytes(stackFolder, entryPoint, []byte(payload.StackFileContent))
-		if err != nil {
-			return nil, httperror.InternalServerError("Unable to persist updated Compose file on disk", err)
-		}
-
-		tempManifestPath, err := handler.convertAndStoreKubeManifestIfNeeded(stackFolder, stack.ProjectPath, entryPoint, relatedEndpointIds)
-		if err != nil {
-			return nil, httperror.InternalServerError("Unable to convert and persist updated Kubernetes manifest file on disk", err)
-		}
+	stack.NumDeployments = len(relatedEndpointIds)
 
-		manifestPath = tempManifestPath
-	}
+	stack.UseManifestNamespaces = payload.UseManifestNamespaces
 
-	if deploymentType == portainer.EdgeStackDeploymentKubernetes {
-		if manifestPath == "" {
-			manifestPath = filesystem.ManifestFileDefaultName
-		}
+	stack.EdgeGroups = groupsIds
 
-		_, err = handler.FileService.StoreEdgeStackFileFromBytes(stackFolder, manifestPath, []byte(payload.StackFileContent))
+	if payload.UpdateVersion {
+		err := handler.updateStackVersion(stack, payload.DeploymentType, []byte(payload.StackFileContent), "", relatedEndpointIds)
 		if err != nil {
-			return nil, httperror.InternalServerError("Unable to persist updated Kubernetes manifest file on disk", err)
+			return nil, httperror.InternalServerError("Unable to update stack version", err)
 		}
 	}
 
-	err = tx.EdgeStack().UpdateEdgeStackFunc(stack.ID, func(edgeStack *portainer.EdgeStack) {
-		edgeStack.NumDeployments = len(relatedEndpointIds)
-		if payload.UpdateVersion {
-			edgeStack.Status = make(map[portainer.EndpointID]portainer.EdgeStackStatus)
-			edgeStack.Version++
-		}
-
-		edgeStack.UseManifestNamespaces = payload.UseManifestNamespaces
-
-		edgeStack.DeploymentType = deploymentType
-		edgeStack.EntryPoint = entryPoint
-		edgeStack.ManifestPath = manifestPath
-
-		edgeStack.EdgeGroups = groupsIds
-	})
+	err = tx.EdgeStack().UpdateEdgeStack(stack.ID, stack)
```

Comment:
This would re-introduce the data-race if the feature flag gets enabled, we need to unfold it into two cases, `UpdateEdgeStackFunc()` when the feature flag is on and `UpdateEdgeStack()` when it's off, that's until the feature flag gets removed, of course.

Reply:
the problem would be with updateStackVersion which stores the file if needed and returns an error. we also did the updateStackVersion think on EE. I'll open a bug about it and we will fix it separately